### PR TITLE
feat: refactor divider

### PR DESCRIPTION
**Pull request by tanvibhakta**
## Description

Fixes https://github.com/appsmithorg/appsmith/issues/19337
Zeplin: https://zpl.io/RMvjvvg

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manual on storybook 

### Test Plan
https://github.com/appsmithorg/TestSmith/issues/2161

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### QA activity:
- [x] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: packages/design-system-v2/src/Divider/Divider.styles.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  --divider-length: 100%;
+`;
+
+export const StyledDivider = styled.span<{
+  props: DividerProps;
+}>`
+  ${Variables}
+
+  /* Base style */
+  background-color: var(--ads-v2-divider-color);
```
Comment (albinAppsmith) [inline packages/design-system-v2/src/Divider/Divider.styles.tsx:17]: Where is this variable coming from?

File: packages/design-system/src/Divider/Divider.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        alignItems: "center",
+        justifyContent: "center",
+      }}
+    >
+      <Divider {...args} />
+    </div>
+  );
+};
+
+export const DividerExample = Template.bind({});
```
Comment (albinAppsmith) [inline packages/design-system/src/Divider/Divider.stories.tsx:36]: DividerExample -> DividerStory

File: packages/design-system/src/Divider/Divider.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  --divider-length: 100%;
+`;
+
+export const StyledDivider = styled.span<{
+  props: DividerProps;
+}>`
+  ${Variables}
+
+  /* Base style */
+  background-color: var(--ads-v2-color-bg-emphasis);
```
[the remark this answers is not in the document]
Comment (tanvibhakta) [reply packages/design-system/src/Divider/Divider.styles.tsx:17]: why? It does not need to be controlled from within the component.

File: packages/design-system/src/Divider/Divider.types.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
+
+export type DividerProps = {
+
+  /** control the direction of the divider*/
+  orientation: "horizontal" | "vertical";
+  /** (try not to) pass addition classes here */
+  className?: string;
+
+}
```
Comment (albinAppsmith) [inline packages/design-system/src/Divider/Divider.types.ts:10]: Extend HTML span props

**Review by albinAppsmith — APPROVED** [review]

**Review by albinAppsmith — APPROVED** [review]

**Comment by shadabbuchh** [comment]
@tanvibhakta just a suggestion (not a bug) can we please keep the default state of the divider a little more enlarged? It looks too small when compared to the size of the canvas.

**Comment by tanvibhakta** [comment]
> the default state of the divider

The divider takes the size of the container, so sure I can change the story for the divider to be bigger! Note that this won't change the underlying component at all.

**Comment by shadabbuchh** [comment]
Looks much better now. This PR is good to go.

**Pull request merged**